homekit/StateManager.js

```diff
@@ -48,14 +48,16 @@ module.exports = (device, platform) => {
 	const log = platform.log
 
 	function updateClimateReact() {
-		device.state.smartMode.enabled = device.state.active && device.state.mode !== 'AUTO'
-		device.state.smartMode.type = 'temperature'
-		device.state.smartMode.highTemperatureWebhook = null
-		device.state.smartMode.lowTemperatureWebhook = null
+		const smartModeState = device.state.smartMode
```

Comment:
If one queries Sensibo's API for smartMode before it was ever set (in the lifetime of the device, I assume this resets after resetting the device though) then this will be an undefined value. This was actually was causing an exception in the v2.3.4 (which I fixed by replacing it with a dummy object that has enabled=false). 

However, once smartMode has been set, even once, it will always be populated (though it could be disabled) - the values will always be there.

Since that's the case, I don't think we need to do any special checks here for this scenario.

Re running updateClimateReact() only when "enableClimateReactAutoSetup" is true, we can do that but that's really a micro optimization because all this function does is update the internal state of smartMode. This is never propagated to the server side (i.e. Sensibo's API) unless "enableClimateReactAutoSetup". On the flip side, adding this check is pretty easy (just return immediately if if disabled) but hints the person reading the code that this might have an actual effect if even disabled, which is not true. I'm open to either so up to you.